config/log_config.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/logging"
+)
+
+// LogPatternConfig is an instance of a level specification for a given logger.
+type LogPatternConfig struct {
```

Comment:
minor, just to make it clear this ultimately parses loggers not logs

```suggestion
type LoggerPatternConfig struct {
```